### PR TITLE
Added Sieve Of Eratosthenes [Golang]

### PR DESCRIPTION
**Pull request by mohitkyadav**
#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

**Comment by aviaryan** [comment]
@mohitkyadav Are you using `golint`? It suggests some style changes in your code.

**Comment by mohitkyadav** [comment]
No, I'm not using `golint`. Thanks for the suggestion I'll try `golint.`

File: sieve_of_eratosthenes/sieve_of_eratosthenes.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+        "fmt"
```
Comment (aviaryan) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: You are using spaces here, and that too 8.

Comment (aviaryan) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: Try using the GoSublime Sublime Text plugin. You might have some problem configuring it, so here is my GoSublime user settings file.
https://gist.github.com/aviaryan/3e2ac784609e046441ee9ba9032baf17

Comment (mohitkyadav) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: Done

File: sieve_of_eratosthenes/sieve_of_eratosthenes.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// SieveOfEratosthenes returns bool array primes
+func SieveOfEratosthenes() [N + 1]bool {
+	var primes [N + 1]bool
+	sqrtOfn := int(math.Floor(math.Sqrt(N)) + 1)
+	for j := range primes {
+		primes[j] = true
+	}
+	for i := 2; i <= sqrtOfn; i++ {
+		if primes[i] {
+			for j := 2 * i; j < N+1; j += i {
```
Comment (mohitkyadav) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: GoSublime is removing whitespace around the binary operator, is it okay?

Comment (aviaryan) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: Yeah, it should be ok. Giving space around every binary operator isn't a good thing to do either. At some point, one must prefer common sense over a hard rule.

File: sieve_of_eratosthenes/sieve_of_eratosthenes.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"fmt"
+	"math"
+)
+
+// N = Limit, change N to get all primes <= N
+const N = 20
+
+// SieveOfEratosthenes returns bool array primes
+func SieveOfEratosthenes() [N + 1]bool {
```
Comment (aviaryan) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: You are taking N as a const. But suppose someone wants to use your code in a program. Don't you think making `n` a normal function variable will be better?

File: sieve_of_eratosthenes/sieve_of_eratosthenes.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+				primes[j] = false
+			}
+		}
+	}
+	return primes
+}
+
+func main() {
+	var primes [N + 1]bool
+	primes = SieveOfEratosthenes()
```
Comment (aviaryan) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: No advantage of declaring primes specifically here. Use `primes := SieveOfEratosthenes()`

File: sieve_of_eratosthenes/sieve_of_eratosthenes.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	"math"
+)
+
+// N = Limit, change N to get all primes <= N
+const N = 20
+
+// SieveOfEratosthenes returns bool array primes
+func SieveOfEratosthenes() [N + 1]bool {
+	var primes [N + 1]bool
+	sqrtOfn := int(math.Floor(math.Sqrt(N)) + 1)
```
Comment (aviaryan) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: As we are following camelCase, it should be sqrtOfN.

File: sieve_of_eratosthenes/sieve_of_eratosthenes.go (hunk 3 above)
Comment (mohitkyadav) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: Done

File: sieve_of_eratosthenes/sieve_of_eratosthenes.go (hunk 4 above)
Comment (mohitkyadav) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: Changed.

File: sieve_of_eratosthenes/sieve_of_eratosthenes.go (hunk 5 above)
Comment (mohitkyadav) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.go]: Changed

**Pull request merged**